Problem building with Piranha

I'm putting together a package for the [Spack package manager](https://github.com/LLNL/spack), and have hit a problem when building with the Piranha CAS library. I'm not quite sure whether this is a symengine problem or a piranha problem. I've attached the [build log](https://github.com/symengine/symengine/files/504445/spack-build.txt) for your perusal, with the salient lines being
```
/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-3YZfHb/symengine/symengine/polys/uintpoly_piranha.h:217:19: error: expected unqualified-id
        static Cf PZERO(0);
                  ^
/usr/include/sys/param.h:124:15: note: expanded from macro 'PZERO'
#define PZERO   22              /* No longer magic, shouldn't be here.  XXX */
```

--- a/bin/install_travis.sh
+++ b/bin/install_travis.sh
@@ -50,7 +50,6 @@ if [[ "${TRAVIS_OS_NAME}" != "osx" ]]; then
         sudo apt-get install cmake libgmp-dev
     fi
 else
-    brew install cmake
     wget https://raw.githubusercontent.com/symengine/dependencies/6a42d290071921a0a478c6883fc0ddd709d664c9/gmp-6.0.0a.tar.bz2
     tar -xjf gmp-6.0.0a.tar.bz2;
     cd gmp-6.0.0 && ./configure --prefix=$our_install_dir --enable-cxx && make -j8 install && cd ..;

--- a/symengine/polys/uintpoly_piranha.h
+++ b/symengine/polys/uintpoly_piranha.h
@@ -214,12 +214,12 @@ public:
 
     const Cf &get_coeff_ref(unsigned int x) const
     {
-        static Cf PZERO(0);
+        static Cf pzero(0);
 
         term temp = term(0, pmonomial{x});
         auto it = this->poly_._container().find(temp);
         if (it == this->poly_._container().end())
-            return PZERO;
+            return pzero;
         return it->m_cf;
     }
 

--- a/symengine/polys/uintpoly_piranha.h
+++ b/symengine/polys/uintpoly_piranha.h
@@ -112,11 +112,14 @@ using pratpoly = piranha::polynomial<rational_class, pmonomial>;
 template <typename Cf, typename Container>
 class PiranhaForIter
 {
-    typename Container::container_type::const_iterator ptr_;
+public:
+    typedef decltype(std::declval<Container &>()._container().begin()) ptr_type;
+
+private:
+    ptr_type ptr_;
 
 public:
-    PiranhaForIter(typename Container::container_type::const_iterator ptr)
-        : ptr_{ptr}
+    PiranhaForIter(ptr_type ptr) : ptr_{ptr}
     {
     }
 

